### DISCUSSION TITLE
How do I add custom js to my header?

### DISCUSSION THREAD

**Discussion by rushishelat** [holger1411/understrap#1321] [Q&A] [answered]
I want to create a sticky navbar underneath my header, but I want the navbar to become sticky once I scroll past my header and to the navbar. I know that I have to use wpb enque in the functions.php folder in the under strap them, however I don’t know where to put the actual js code? Also do I need to do anything with my gulp files? If someone can explain the steps to do this I would really appreciate it. Thanks!

**Comment by danemorgan** [comment, marked as answer]
First, if you want to do this and be able to follow updates to the main theme, you need to be doing this in a child theme. We have a base child theme to start from here https://github.com/understrap/understrap-child

Either way there will be a /js/ folder in the theme folder which is there to hold JavaScript files. Following that you can enqueue the file as normal in your functions.php file. Depending on the library you are using you will likely need to add some small inline script to your header.php file following the menu.

**Reply by rushishelat** [reply]
**I am now working on the child theme and I enqued the following into my funtions.php:** 

function custom_js() {
    wp_enqueue_script( 'custom_js', get_stylesheet_directory_uri() . 'js/custom-javascript.js', array( 'jquery' ), '1.0', false );
}

add_action('wp_enqueue_scripts', 'custom_js');

**I created a js file in my js folder called custom-javascript.js and added the following code to the file:**

window.onscroll = function() {myFunction()};

// Get the navbar
var navbar = document.getElementById("wrapper-navbar");

// Get the offset position of the navbar
var sticky = navbar.offsetTop;

// Add the sticky class to the navbar when you reach its scroll position. Remove "sticky" when you leave the scroll position
function myFunction() {
  if (window.pageYOffset >= sticky) {
    navbar.classList.add("sticky")
  } else {
    navbar.classList.remove("sticky");
  }
}

**In my _child_theme.scss file i added the following code:**

.content{
    padding: 16px;
}

.sticky {
    position: fixed;
    top: 0;
    width: 100%;
  }

.sticky + .content{
    padding-top: 60px;
}

**I added this code to test out the onscroll method on the starter template of the child theme. So there is no customization, i just want to get the website to respond to the changes i am making with js. Unfortunately, nothing is happening. I would really appreciate your help in figuring out how to get my custom-javascript.js file to influence my webpage. Thank you in advance!**

**Reply by rushishelat** [reply]
Nevermind i got it. Thank you for your advice!
